### PR TITLE
whitelist wildcard

### PR DESCRIPTION
**Pull request by phynias**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [ X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [ X] I have made only one major change in my proposed changes.
- [ X] I have commented my proposed changes within the code.
- [ X] I have tested my proposed changes, and have included unit tests where possible.
- [ X] I am willing to help maintain this change if there are issues with it later.
- [ X] I give this submission freely and claim no ownership.
- [ X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
*A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues*
add whitelist wildcards 

**How does this PR accomplish the above?:**
*A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix*
new file with domain names you want to remove

**What documentation changes (if any) are needed to support this PR?:**
*A detailed list of any necessary changes*
Would need a section on manually adding domains to a new file since this does not have anythign in the web admin yet.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

File: gravity.sh [hunk 1]
```diff
@@ -447,6 +448,14 @@ gravity_Whitelist() {
   str="Number of whitelisted domains: ${num}"
   echo -ne "  ${INFO} ${str}..."
 
+  if [[ -f "${whitelistWild}" ]]; then
+    while read p
```
[the remark this answers is not in the document]
Comment (phynias) [reply gravity.sh]: sorry @Mcat12 not sure what you mean. maybe if you give an example?

**Comment by phynias** [comment]
sorry @Mcat12 not sure what you mean. maybe if you give an example?

**Comment by Mcat12** [comment]
If you add `example.com$` to `/etc/pihole/regex.list` and reload FTL so it rereads the config, `sudo service pihole-FTL restart`, this will not prevent `test.example.com` from being blocked.

Also, in the current v3.3.1 release, this will not prevent a wildcard blacklisted domain from being blocked (add `example.com` to the wildcard blacklist).

**Comment by phynias** [comment]
so you're saying if someone both whitelists and blacklists something? if someone does both i think it should end up blacklisting it. at the end of the day this is meant for blocking things. it would make more sense to add the checking of this in the web ui than here. 

add a domain to  wildcard whitelist -> first thing to check is if it's blacklisted -> show error.

**Comment by Mcat12** [comment]
Fair point, but even so the PR needs more work to actually implement the integration with the rest of the program.

**Comment by phynias** [comment]
more being the web ui part? if so i can work on that and getit back here soon. if it's other things just let me know.

thanks.

**Comment by Mcat12** [comment]
The core scripts (`pihole -w` for example) need to be updated.

**Comment by DL6ER** [comment]
Wildcard whitelisting will be implemented directly in *FTL*DNS via https://github.com/pi-hole/FTL/pull/299

**Pull request closed without merging**